src/lib/Behat/EnvironmentRestore.php

```diff
@@ -0,0 +1,125 @@
+
+    /** @var array Names of available installer services in Studio */
+    private $installerServices = ['platform' => 'ezplatform.installer.clean_installer',
+        'platform-demo' => 'app.installer.demo_installer',
+        'platform-ee' => 'ezstudio.installer.studio_installer',
+        'platform-ee-demo' => 'app.installer.ee_demo_installer', ];
+
+    /**
+     * EnvironmentRestore constructor.
+     *
+     * @param \Symfony\Component\DependencyInjection\ContainerInterface $serviceContainer
+     */
+    public function __construct(ContainerInterface $serviceContainer)
+    {
+        $this->serviceContainer = $serviceContainer;
+    }
+
+    /**
+     * Detects installation type and restores the database using parameters from parameters.yml.
+     */
+    public function restoreDatabase(): void
+    {
+        $currentDirectory = getcwd();
+        $installer = $this->getInstallerService();
+        $this->findDirectoryToRunScripts();
+        $installer->setOutput(new NullOutput());
+        $installer->importSchema();
+        $installer->importData();
+        chdir($currentDirectory);
+    }
+
+    /**
+     * Clears cache.
+     *
+     * Override cache path in AppKernel (getCacheDir method) to point to another directory
+     */
+    public function clearCache(): void
+    {
+        $pool = $this->serviceContainer->get('ezpublish.cache_pool');
+        $pool->clear();
+    }
+
+    /**
+     * Traverses up the directory tree until a directory where database scripts should be run is found.
+     *
+     * @throws \Exception When the directory is not found going up to 5 levels
+     */
+    private function findDirectoryToRunScripts(): void
+    {
+        $counter = 0;
+        while (!$this->areExpectedDirectoriesPresent()) {
+            chdir('..');
+            if ($counter > 4) {
+                throw new \Exception('Directory to run restoring scripts was not found!');
+            }
+
+            ++$counter;
+        }
+    }
+
+    /**
+     * Verifies whether the current folder contains the expected directories.
+     *
+     * @return bool
+     */
+    private function areExpectedDirectoriesPresent(): bool
+    {
+        $actualDirectories = scandir(getcwd(), 0);
+
+        $presentExpectedDirectories = array_intersect($this->expectedDirectories, $actualDirectories);
+        sort($this->expectedDirectories);
+        sort($presentExpectedDirectories);
+
+        return $this->expectedDirectories === $presentExpectedDirectories;
+    }
+
+    /**
+     *  Returns installer service depending on current install type.
+     *
+     * @throws \Symfony\Component\DependencyInjection\Exception\ServiceCircularReferenceException
+     * @throws \Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException  When no suitable service is found
+     *
+     * @return Installer Installer service
+     */
+    private function getInstallerService(): Installer
+    {
```

Comment:
I could use foreach here, but this way puts more emphasis on the order (and the order is important).